### PR TITLE
#178601157 allow admin change user referral percentage

### PR DESCRIPTION
**Pull request by tobieeniafe**
#### What does this PR do?
Allow admin change user referral percentage

#### Description of Task to be completed?
Allow admin change user referral percentage

#### How should this be manually tested?
`PUT /api/v1/referral/percentage`
#### What are the relevant pivotal tracker stories?
#178601157

### CODE REVIEW

**Review by andela-hpopoola — CHANGES_REQUESTED** [review]
Thank you for working on this. The referral percentage should be in the user model as discussed in the morning. I am dropping this comment just to be sure that we are on the same page. Thank you.

File: server/models/user.model.js [hunk 1]
```diff
@@ -107,6 +107,9 @@ const UserSchema = new mongoose.Schema(
           type: String,
         },
       },
+      referralPercentage: {
+        type: Number,
```
Comment (andela-hpopoola) [inline server/models/user.model.js:111]: Can we make the default the current REFERRAL_PERCENTAGE for consistency?

File: test/controllers/user.controller.test.js [hunk 2]
```diff
@@ -3494,6 +3494,80 @@ describe('User Controller', () => {
         useExistingUser: true,
       });
     });
+
+    describe('Update user referral percentage', () => {
+      const method = 'put';
+      const endpoint = '/api/v1/user/referral-percentage';
+
+      const vendorUser = UserFactory.build(
+        { role: USER_ROLE.VENDOR, activated: true, 'additionalInfo.referralPercentage': 1.5 },
```
Comment (andela-hpopoola) [inline test/controllers/user.controller.test.js:3503]: Lets add the default referral percentage value of 1.5  to the user factory.

**Review by andela-hpopoola — APPROVED** [review]
Looks good to me. Thank you for working on this.

File: test/controllers/user.controller.test.js (hunk 2 above)
Comment (tobieeniafe) [reply test/controllers/user.controller.test.js:3503]: Adding it to the factory breaks the tests because of Joi.

**Pull request merged**